### PR TITLE
Download metrics impr

### LINKED ISSUE #193: Accept date in URL query for DownloadMetrics

### PR DESCRIPTION
**Pull request by jendakol**
1. Accept "all" as date
2. Accept date both in payload and URL query

Closes #193

### CODE REVIEW

File: internal/functions/metricsapi/download.go [hunk 1]
```diff
@@ -12,33 +12,59 @@ import (
...
 	var ctx = r.Context()
-	logger := logging.FromContext(ctx).Named("DownloadMetrics")
 
 	client := store.Client{}
 	date := time.Now()
 
+	downloadMetrics(ctx, w, r, client, date)
+}
+
+func downloadMetrics(ctx context.Context, w http.ResponseWriter, r *http.Request, client store.Client, date time.Time) {
```
Comment (dizider) [inline internal/functions/metricsapi/download.go:31]: I am not sure if I understand it well. When I want todays metric, I need to provide current date in request?

Comment (jendakol) [reply internal/functions/metricsapi/download.go:31]: Nope. There's a fallback when no date can be parsed from the request, today's date is used. That's the `date` param provided.

**Review by dizider — APPROVED** [review]
just one misunderstanding how to get today's metrics

File: internal/functions/metricsapi/download.go (hunk 1 above)
Comment (vitkolos) [reply internal/functions/metricsapi/download.go:31]: @dizider you can but don't have to. Today's date is the default.

**Pull request merged**